docs/rules/no-callback-in-promise.md

```diff
@@ -1,3 +1,55 @@
-# Avoid calling `cb()` inside of a `then()` (use [nodeify][] instead) (no-callback-in-promise)
+# Avoid calling `cb()` inside of a `then()` or `catch()` (no-callback-in-promise)
+
+As a general rule, callbacks should never be directly invoked inside a [Promise.prototype.then()] or [Promise.prototype.catch()] method. That's because your callback may be unintentionally be invoked twice. Take the following example:
+
+```js
+function callback(err, data) {
+  console.log("Callback got called with:", err, data);
+  throw new Error("My error");
+}
+
+// note: passing `err.message` for demo purposes, normally you would pass `err`
+Promise.resolve()
+  .then(() => callback(null, "data"))
+  .catch(err => callback(err.message, null));
+```
+
+If you run this example, your output will look like the following:
+
+```
+Callback got called with: null data
+Callback got called with: My error null
+```
+
+**How to fix it?**
+
+Ensure that your callback invocations are wrapped by a deferred execution function such as:
+- [setImmediate()] or [process.nextTick()]: for Node.js.
+- [setTimeout()]: for Browsers and Node.js.
+
+```js
+// node.js
+Promise.resolve()
+  .then(() => setImmediate(() => callback(null, "data")))
```

Comment:
while this works, i don't love these either, and would prefer that folks `promisify`'d their code instead of doing this